slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/dao/MetadataDao.java

```diff
@@ -36,4 +36,32 @@ public void insertGenesisValidatorsRoot(
         .bind(1, genesisValidatorsRoot)
         .execute();
   }
+
+  public Optional<HighWatermark> findHighWatermark(Handle handle) {
+    return handle
+        .createQuery(
+            "SELECT high_watermark_epoch as epoch, high_watermark_slot as slot FROM metadata WHERE id = ?")
+        .bind(0, METADATA_ROW_ID)
+        .mapToBean(HighWatermark.class)
+        .filter(h -> h.getEpoch() != null && h.getSlot() != null)
+        .findFirst();
+  }
+
+  public int updateHighWatermark(final Handle handle, final HighWatermark highWatermark) {
```

Comment:
Does this need a constraint on the high watermark being height then all of the low watermarks?

Reply:
I didn't add anything around that because this AC simplifies it:
> Web3signer operator is responsible for setting a good high-watermark that is in the future 

...but: even though you could still set a HWM such that LWM < HWM < latestSignedData, it's probably not a bad (and cheap) check to have in place.